src/main/java/com/diffmin/patch/PatchApplication.java

```diff
@@ -77,6 +78,28 @@ private static void performInsertion(
                                 ((CtExecutable<?>) toBeInserted.getParent()).getThrownTypes());
                 ((CtExecutable<?>) inWhichElement).setThrownTypes(thrownTypesCopy);
                 break;
+            case CONTAINED_TYPE:
+                // Inserting into CtPackage
+                Set<CtType<?>> typesCopy =
+                        ((CtType<?>) toBeInserted)
+                                .getPackage().getTypes().stream()
+                                        .map(CtType::clone)
+                                        .collect(Collectors.toSet());
+                ((CtPackage) inWhichElement).setTypes(typesCopy);
```

Comment:
So, the comment I had here before was not supposed to be posted, I hadn't finished thinking. Anyway, this is wrong (as was the previous version), as it replaces _all_ of the types in the package. You only want to add the new type, not replace existing ones (as that renders everything else diffmin does completely redundant, you've already replaced the entire type!)

Reply:
> as it replaces all of the types in the package

For our case, the package would be a single file (the PREV one), right?

> You only want to add the new type, not replace existing ones

I agree but this wasn't possible because types in `CtPackage` are stored in a `HashSet` so I don't have any control over the position of insertion which, apparently, is important. Thus, I had to set the entire types at once. This workaround is similar to the one we have implemented for insertion of `ThrownType`s. There will be redundant operations and we have to resolve that later. 

`CtCompilationUnit` stores its types in a list and in that case, we were able to insert a `CtType` at a precise location.